File: backend/src/main/java/com/example/backend/auth/dto/AuthResponse.java
package com.example.backend.auth.dto;

import com.example.backend.user.entity.Role;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AuthResponse {
    private Long userId;
    private String email;
    private Role role;
    private String firstName;
    private String lastName;
    private Boolean isVerified;
    private Boolean isApproved;
    private String message;
}

File: frontend/lib/auth/models/auth_response.dart
import 'user.dart';
import 'role.dart';

class AuthResponse {
  final int userId;
  final String email;
  final Role? role;
  final String firstName;
  final String lastName;
  final bool? isVerified;
  final bool? isApproved;
  final String message;

  AuthResponse({
    required this.userId,
    required this.email,
    this.role,
    required this.firstName,
    required this.lastName,
    this.isVerified,
    this.isApproved,
    required this.message,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) {
    return AuthResponse(
      userId: json['userId'] ?? 0,
      email: json['email'] ?? '',
      role: Role.fromString(json['role']?.toString()),
      firstName: json['firstName'] ?? '',
      lastName: json['lastName'] ?? '',
      isVerified: json['isVerified'],
      isApproved: json['isApproved'],
      message: json['message'] ?? '',
    );
  }

  User toUser() {
    return User(
      id: userId,
      email: email,
      firstName: firstName,
      lastName: lastName,
      role: role,
      isVerified: isVerified,
      isApproved: isApproved,
    );
  }
}
